internal/resources/metal/metal_connection/speed.go

```diff
@@ -0,0 +1,45 @@
+package metal_connection
+
+import (
+	"fmt"
+	"strings"
+)
+
+var (
+	Mega          uint64 = 1000 * 1000
+	Giga          uint64 = 1000 * Mega
+	AllowedSpeeds        = []struct {
```

Comment:
Why is `AllowedSpeeds` public? It's not used in the test.

Reply:
IMO they should either be all private or all public; as you point out they were only made public because they are referenced in a test.  Since the test that references them is a unit test I've instead opted to keep the test in the same package as the code so that these can remain private; this PR has been updated with those changes.